packages/ember-glimmer/tests/integration/syntax/each-in-test.js

```diff
@@ -0,0 +1,142 @@
+
+  get truthyValue() { return { 'Not Empty': 1 }; }
+  get falsyValue() { return {}; }
+
+}
+
+applyMixins(EachInTest,
+
+  SyntaxCondtionalTestHelpers,
+
+  new TruthyGenerator([
+    // TODO: figure out what the rest of the cases are
+    { foo: 1 }
+  ]),
+
+  new FalsyGenerator([
+    // TODO: figure out what the rest of the cases are
+    {},
+    Object.create({ 'Not Empty': 1 }),
+    undefined,
+    null
+  ])
+
+  // TODO(mmun): Add support for object proxies and
+  // include the ObjectTestCases mixin.
+);
+
+moduleFor('@htmlbars Syntax test: {{#each-in}}', class extends EachInTest {
+
+  templateFor({ cond, truthy, falsy }) {
+    return `{{#each-in ${cond}}}${truthy}{{else}}${falsy}{{/each-in}}`;
+  }
+
+  [`@test it repeats the given block for each item in the hash`]() {
+    this.render(strip`
+      <ul>
+        {{#each-in categories as |category count|}}
+          <li>{{category}}: {{count}}</li>
+        {{/each-in}}
+      </ul>
+    `, {
+      categories: {
+        'Smartphones': 8203,
+        'JavaScript Frameworks': Infinity
+      }
+    });
+
+    this.assertStableRerender();
+
+    this.assertHTML(strip`
+      <ul>
+        <li>Smartphones: 8203</li>
+        <li>JavaScript Frameworks: Infinity</li>
+      </ul>
+    `);
+
+    this.runTask(() => set(this.context, 'categories', {
+      'Smartphones': 100
+    }));
+
+    this.assertHTML(`<ul><li>Smartphones: 100</li></ul>`);
+
+    this.runTask(() => set(this.context, 'categories', {
+      'Programming Languages': 199303,
+      'Good Programming Languages': 123,
+      'Bad Programming Languages': 456
+    }));
+
+    this.assertHTML(strip`
+      <ul>
+        <li>Programming Languages: 199303</li>
+        <li>Good Programming Languages: 123</li>
+        <li>Bad Programming Languages: 456</li>
+      </ul>
+    `);
+  }
+
+  [`@test it only iterates over an object's own properties`]() {
+    let protoCategories = {
+      'Smartphones': 8203,
+      'JavaScript Frameworks': Infinity
+    };
+
+    let categories = Object.create(protoCategories);
+    categories['Televisions'] = 183;
+    categories['Alarm Clocks'] = 999;
+
+    this.render(strip`
+      <ul>
+        {{#each-in categories as |category count|}}
+          <li>{{category}}: {{count}}</li>
+        {{/each-in}}
+      </ul>
+    `, { categories });
+
+    this.assertStableRerender();
+
+    this.assertHTML(strip`
+      <ul>
+        <li>Televisions: 183</li>
+        <li>Alarm Clocks: 999</li>
+      </ul>
+    `);
+
+    this.runTask(() => set(this.context, 'categories', {
+      'Programming Languages': 199303,
+      'Good Programming Languages': 123,
+      'Bad Programming Languages': 456
+    }));
+
+    this.assertHTML(strip`
+      <ul>
+        <li>Programming Languages: 199303</li>
+        <li>Good Programming Languages: 123</li>
+        <li>Bad Programming Languages: 456</li>
+      </ul>
+    `);
+
+    this.runTask(() => set(this.context, 'categories', categories));
```

Comment:
Can you make a new `categories` object here for the replacement step?

Something like:

```javascript
categories = Object.create(protoCategories);
categories['Televisions'] = 183;
categories['Alarm Clocks'] = 999;
```